README.md

```diff
@@ -52,3 +52,25 @@ etc.). There is no implicit conversion to Michelson's binary format (`PACK`) nor
 quoting mechanism.
 
 After the conversion, the bytes can be use in the field `metadata` when originating the contract.
+
+## Error Code
+
+In error scenarios, the contract fails with an error code.
+
+Here is a summary of all the error codes thrown by the contract.
+(The list of errors may be inaccurate and incomplete, it will be updated during the implementation.)
+
+| Error Code       | Description                                                                                                              |
+|------------------|--------------------------------------------------------------------------------------------------------------------------|
+| 100              | Thrown by `get_tick`: Cannot get tick in tick indexed state, tick is not initialized.                                    |
+| 101              | Thrown by `initialize_tick`: Invalid withness.                                                                           |
+| 102              | Thrown by `collect_fees`: Position does not exist.                                                                       |
+| 103              | Thrown by `floor_log_half_bps`: Log out of bounds.                                                                       |
+| 104              | Thrown by `assert_nat`: X should be positive.                                                                            |
+| 105              | Internal Contract Error (Impossible state reached).                                                                      |
+| 106              | Thrown by `half_bps_pow_rec`: Should not reach end of ladder.                                                            |
+| 107              | Thrown by `x_to_y` and `y_to_x`: Past deadline.                                                                          |
+| 108              | Thrown by `x_to_y`: `dy` received < `min_dy`.                                                                            |
+| 109              | Thrown by `y_to_x`: `dx` received < `min_dx`.                                                                            |
+| 110              | Trhown by `x_transfer` and `y_transfer`: Invalid entrypoint for the contract transfer.                                   |
```

Comment:
@Martoon-00 @dcastro This MR is ready for review, the description may be still lacking, hope to get your feedbacks as well.

Reply:
I've updated this to `Trying to transfer assert x or y to an entrypoint that does not exist.` Let me know if this is good enough.

Reply:
"transfer to an entrypoint" sounds a bit odd, though this just comes from the fact that addresses (that we, semantically, transfer to) and entrypoints (that we call) are coupled in the code.

To make this clearer for an arbitrary reader, maybe say "There is no appropriate `transfer` entrypoint in the provided token contract", or smth like that?

Reply:
Make sense, I've updated the description.